### PR TITLE
Add contribution tracking for sddr materials

### PR DESCRIPTION
**Pull request by dodu94**
# Description

Should be reviewed after PR #300 and PR #306 

This PR adds the tracking of the contribution to the SDDR tally of the different parents to the typical materials assessed in the SphereSDDR benchmark. The implementation is similar to what has been done for the FNG SDDR benchmark.

## Type of change

Please select what type of change this is.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New benchmark
    - [ ] Non-breaking change which entirely uses exisiting classes, structure etc
    - [ ] Breaking change which has implemented new/modified classes etc
- [x] New feature 
    - [x] Non-breaking change which adds functionality
    - [ ] Breaking change fix or feature that would cause existing functionality to not work as expected

## Other changes

- [ ] This change requires a documentation update
- [ ] (If Benchmark) This requires additional data that can be obtained from:
    - Benchmark data 1
    - Benchamrk data 2

# Testing

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Some additional tests have been added to the CI suite and the feature has also been tested in a production case.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] General testing 
    - [x] New and existing unit tests pass locally with my changes
    - [x] Coverage is >80%

### CODE REVIEW

File: docs/source/requirements.txt
Comment (alexvalentine94) [inline docs/source/requirements.txt:1]: Just check this still holds as some time since PR submitted.

File: setup.cfg
Comment (alexvalentine94) [inline setup.cfg:1]: also to check

**Review by alexvalentine94 — COMMENTED** [review]
These changes look good to me. The additions have also been well tested with the implementation of new tests. I just want to check are still happy with the minimal version of F4Enix that can be supported since it has been some time since the PR was submitted.

File: docs/source/requirements.txt
Comment (dodu94) [reply docs/source/requirements.txt:1]: I believe each PR sets its minimum version of F4Enix (if needed). If they are approved in the correct order it should not be an issue.

File: setup.cfg
Comment (dodu94) [reply setup.cfg:1]: I believe each PR sets its minimum version of F4Enix (if needed). If they are approved in the correct order it should not be an issue.

**Review by alexvalentine94 — APPROVED** [review]

**Pull request merged**